src/file_analysis/AnalyzerSet.cc

```diff
@@ -145,10 +155,15 @@ file_analysis::Analyzer* AnalyzerSet::InstantiateAnalyzer(const Tag& tag, Record
     return a;
 }
 
-void AnalyzerSet::Insert(file_analysis::Analyzer* a, std::unique_ptr<zeek::detail::HashKey> key) {
+void AnalyzerSet::Insert(file_analysis::Analyzer* a, const Tag& tag, RecordValPtr args) {
     DBG_LOG(DBG_FILE_ANALYSIS, "[%s] Add analyzer %s", file->GetID().c_str(),
             file_mgr->GetComponentName(a->Tag()).c_str());
-    analyzer_map.Insert(key.get(), a);
+
+    auto lv = make_intrusive<ListVal>(TYPE_ANY);
+    lv->Append(tag.AsVal());
+    lv->Append(std::move(args));
+
+    analyzer_map.insert({lv, a});
```

Comment:
Instead of `ListVal`, could the key be a `std::pair<zeek::Tag, RecordValPtr>` or `std::tuple` as the key instead?

Though, the fact that `args` is a pointer and can be mutated after having been used as a key into the `analyzer_map` seems dangerous/unexpected. Isn't it?